src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -434,6 +434,18 @@ internal bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter conso
         private bool ReportErrors(DiagnosticBag diagnostics, TextWriter consoleOutput, ErrorLogger errorLoggerOpt)
             => ReportErrors(diagnostics.ToReadOnly(), consoleOutput, errorLoggerOpt);
 
+        internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
```

Comment:
>HasUnsuppressedErrors [](start = 29, length = 21)

It looks like this method is supposed to be called after the filtering is applied to content of the bag. It doesn't look like this happens for every call site.

Reply:
I'm not sure it matters. Since you can't suppress an error that was originally an error, it shouldn't make a difference.

Reply:
@AlekseyTs Did this address your concern?

Reply:
> Did this address your concern?

I do not think it did. If we haven't done filtering, we didn't have a chance to apply warn as error, correct?

---
In reply to: [266674162](https://github.com/dotnet/roslyn/pull/34174#discussion_r266674162) [](ancestors = 266674162)

Reply:
Also, if it doesn't matter at many call sites, as you are saying. Then why adjusting those call sites?

---
In reply to: [266677762](https://github.com/dotnet/roslyn/pull/34174#discussion_r266677762) [](ancestors = 266677762,266674162)

Reply:
That's correct. I guess I might be thinking of a different scenario: my intention was only to use this helper as a "bailout" mechanism to stop compilation, meaning that not catching errors as early as possible may affect how soon compilation stops but not affect semantic guarantees.

Did you find a place where I'm not doing that?